pkg/systemd/overview-cards/insights.jsx

```diff
@@ -0,0 +1,172 @@
+
+export class InsightsStatus extends React.Component {
+    constructor() {
+        super();
+        this.state = { };
+    }
+
+    componentDidMount() {
+        this.insights_client_timer = service.proxy("insights-client.timer");
+        this.insights_client_timer.addEventListener("changed", () => this.setState({}));
+
+        this.hosts_details_file = cockpit.file("/var/lib/insights/host-details.json", { syntax: JSON });
+        this.hosts_details_file.watch(data => this.setState({ host_details: data }));
+        this.insights_details_file = cockpit.file("/var/lib/insights/insights-details.json", { syntax: JSON });
+        this.insights_details_file.watch(data => this.setState({ insights_details: data }));
+    }
+
+    componentWillUnmount() {
+        this.hosts_details_file.close();
+        this.insights_details_file.close();
+    }
+
+    refresh_insights_status() {
+        const subfeats = (cockpit.manifests.subscriptions && cockpit.manifests.subscriptions.features) || { };
+        if (subfeats.insights && this.insights_client_timer.exists && !this.insights_client_timer.enabled)
+            this.setState({ insightsLinkVisible: true });
+        else
+            this.setState({ insightsLinkVisible: false });
+    }
+
+    render() {
+        const subfeats = (cockpit.manifests.subscriptions && cockpit.manifests.subscriptions.features) || { };
+
+        if (!this.insights_client_timer) {
+            // Not mounted yet
+            return null;
+        }
+
+        if (!this.insights_client_timer.exists) {
+            // insights-client is not installed
+            return null;
+        }
+
+        if (!this.insights_client_timer.enabled) {
+            // machine is not registered with Insights
+            if (subfeats.insights) {
+                // subscriptions page can register us
+                return (
+                    <li className="system-health-insights">
+                        <span className="fa fa-exclamation-triangle" />
+                        <a id="insights_text" tabIndex='0' role="button"
+                    onClick={() => cockpit.jump("/subscriptions")}>
+                            {_("Not connected to Insights")}
+                        </a>
+                    </li>);
+            } else
+                return null;
+        }
+
+        let url;
+        try {
+            url = "http://cloud.redhat.com/insights/inventory/" + this.state.host_details.results[0].id + "/insights";
+        } catch (err) {
+            url = "http://cloud.redhat.com/insights";
+        }
+
+        let icon, text;
+        try {
+            const n_rule_hits = this.state.insights_details.length;
+            if (n_rule_hits == 0) {
+                icon = <span className="fa fa-check-circle-o" />;
+                text = _("No rule hits");
+            } else {
+                try {
+                    const n_hits_by_risk = [0, 0, 0, 0];
+                    this.state.insights_details.forEach(r => {
```

Comment:
This computation should move out of the render method.  Then we can also print a warning message when it fails.

Reply:
Done.